### DISCUSSION TITLE
Is there a way to consume the topic and payload from the Publish without copy?

### DISCUSSION THREAD

**Discussion by xudesheng** [ntex-rs/ntex-mqtt#112] [Q&A] [answered]
during handling incoming message: `v5::client::ControlMessage::Publish(publish) `, I can't find a way to consume the `payload` in `pub struct Publish(codec::Publish);` without copy/clone since it's a private field.
Is there a way to take out the payload without copy/clone? 
same question applies to the `topic` field.

**Comment by fafhrd91** [comment, marked as answer]
api need to be changed a bit. i will try to change api this weekeend

**Reply by xudesheng** [reply]
> api need to be changed a bit. i will try to change api this weekeend

Thank you so much for your hard work. This makes me possible to migrate one of my applications to ntex-mqtt in this weekend too.

**Reply by fafhrd91** [reply]
added .into_inner() method

**Reply by fafhrd91** [reply]
0.8.10

**Reply by xudesheng** [reply]
it meets my expectation perfectly! Thanks, Nikolay, have a nice weekend.
